### PR TITLE
Ignore expected grpc error

### LINKED ISSUE #282: Ignore periodically goaway sent from control plane
**What would you like to be added**:

Istiod can set a connection max duration to keep multiple istiod instances balanced. This will make it send goaway to all the clients. 
As a normal client, kmesh should not treat this as an error. 

**Why is this needed**:

It is very confusing with these logs

```
time="2024-04-28T18:51:32Z" level=warning msg="reconnect due to stream recv failed, rpc error: code = Unavailable desc = closing transport due to: connection error: desc = \"error reading from server: EOF\", received prior goaway: code: NO_ERROR, debug data: \"max_age\"" subsys=controller
time="2024-04-28T19:20:11Z" level=warning msg="reconnect due to stream recv failed, rpc error: code = Unavailable desc = closing transport due to: connection error: desc = \"error reading from server: EOF\", received prior goaway: code: NO_ERROR, debug data: \"max_age\"" subsys=controller
time="2024-04-28T19:50:29Z" level=warning msg="reconnect due to stream recv failed, rpc error: code = Unavailable desc = closing transport due to: connection error: desc = \"error reading from server: EOF\", received prior goaway: code: NO_ERROR, debug data: \"max_age\"" subsys=controller
time="2024-04-28T20:18:53Z" level=warning msg="reconnect due to stream recv failed, rpc error: code = Unavailable desc = closing transport due to: connection error: desc = \"error reading from server: EOF\", received prior goaway: code: NO_ERROR, debug data: \"max_age\"" subsys=controller
time="2024-04-28T20:47:05Z" level=warning msg="reconnect due to stream recv failed, rpc error: code = Unavailable desc = closing transport due to: connection error: desc = \"error reading from server: EOF\", received prior goaway: code: NO_ERROR, debug data: \"max_age\"" subsys=controller
time="2024-04-28T21:17:49Z" level=warning msg="reconnect due to stream recv failed, rpc error: code = Unavailable desc = closing transport due to: connection error: desc = \"error reading from server: EOF\", received prior goaway: code: NO_ERROR, debug data: \"max_age\"" subsys=controller
time="2024-04-28T21:47:43Z" level=warning msg="reconnect due to stream recv failed, rpc error: code = Unavailable desc = closing transport due to: connection error: desc = \"error reading from server: EOF\", received prior goaway: code: NO_ERROR, debug data: \"max_age\"" subsys=controller
time="2024-04-28T22:14:58Z" level=warning msg="reconnect due to stream recv failed, rpc error: code = Unavailable desc = closing transport due to: connection error: desc = \"error reading from server: EOF\", received prior goaway: code: NO_ERROR, debug data: \"max_age\"" subsys=controller
time="2024-04-28T22:48:04Z" level=warning msg="reconnect due to stream recv failed, rpc error: code = Unavailable desc = closing transport due to: connection error: desc = \"error reading from server: EOF\", received prior goaway: code: NO_ERROR, debug data: \"max_age\"" subsys=controller
time="2024-04-28T23:17:47Z" level=warning msg="reconnect due to stream recv failed, rpc error: code = Unavailable desc = closing transport due to: connection error: desc = \"error reading from server: EOF\", received prior goaway: code: NO_ERROR, debug data: \"max_age\"" subsys=controller
time="2024-04-28T23:47:06Z" level=warning msg="reconnect due to stream recv failed, rpc error: code = Unavailable desc = closing transport due to: connection error: desc = \"error reading from server: EOF\", received prior goaway: code: NO_ERROR, debug data: \"max_age\"" subsys=controller
time="2024-04-29T00:17:50Z" level=warning msg="reconnect due to stream recv failed, rpc error: code = Unavailable desc = closing transport due to: connection error: desc = \"error reading from server: EOF\", received prior goaway: code: NO_ERROR, debug data: \"max_age\"" subsys=controller
time="2024-04-29T00:47:42Z" level=warning msg="reconnect due to stream recv failed, rpc error: code = Unavailable desc = closing transport due to: connection error: desc = \"error reading from server: EOF\", received prior goaway: code: NO_ERROR, debug data: \"max_age\"" subsys=controller
time="2024-04-29T01:18:36Z" level=warning msg="reconnect due to stream recv failed, rpc error: code = Unavailable desc = closing transport due to: connection error: desc = \"error reading from server: EOF\", received prior goaway: code: NO_ERROR, debug data: \"max_age\"" subsys=controller

```

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
**What type of PR is this?**

/kind cleanup
/kind enhancement

**What this PR does / why we need it**:

Fix the error log. And do some cleanup

**Which issue(s) this PR fixes**:
Fixes #282

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: pkg/controller/client.go [hunk 1]
```diff
@@ -116,47 +118,41 @@ func (c *XdsClient) clientResponseProcess(ctx context.Context) {
...
-				if err = c.AdsStream.AdsStreamProcess(); err != nil {
+				if err = c.AdsStream.HandleAdsStream(); err != nil {
 					_ = c.AdsStream.Stream.CloseSend()
-					_ = c.grpcConn.Close()
-					reconnect = true
-					continue
 				}
 			} else if bpfConfig.WdsEnabled() {
-				if err = c.workloadStream.WorkloadStreamProcess(c.rbac); err != nil {
+				if err = c.workloadStream.HandleWorkloadStream(c.rbac); err != nil {
```
Comment (LiZhenCheng9527) [inline pkg/controller/client.go:130]: Right now kmesh's mode defaults to workload mode. I think it's better to judge WdsEnable first.

File: pkg/controller/workload/workload_event.go [hunk 2]
```diff
@@ -48,7 +48,7 @@ var (
 
 type ServiceEvent struct {
 	ack *service_discovery_v3.DeltaDiscoveryRequest
-	rqt *service_discovery_v3.DeltaDiscoveryRequest
+	req *service_discovery_v3.DeltaDiscoveryRequest
```
Comment (LiZhenCheng9527) [inline pkg/controller/workload/workload_event.go:51]: There is also a misnamed rqt in ads, can we fix it together?

**Comment by hzxuzhonghu** [comment]
@nlgwcy

**Pull request merged**